clang/lib/Driver/Compilation.cpp

```diff
@@ -47,7 +47,9 @@ Compilation::Compilation(const Driver &D, const ToolChain &_DefaultToolChain,
 Compilation::~Compilation() {
   // Remove temporary files. This must be done before arguments are freed, as
   // the file names might be derived from the input arguments.
-  if (!TheDriver.isSaveTempsEnabled() && !ForceKeepTempFiles)
+  if (!(TheDriver.isSaveTempsEnabled() ||
+        TheDriver.isDumpDeviceCodeEnabled()) &&
+      !ForceKeepTempFiles)
```

Comment:
This is causing all of the temporaries to be kept around - even the non-device code files.  The ability to dump the device code probably shouldn't impact the rest of the files involved unless `-save-temps` is given as well.

Reply:
I checked this code and it only saved the .spv files in the user input directory.
It did not save the other temporary files.

Reply:
Is it possible to save all temporary files in a user input directory ? Thanks.

Reply:
@jinz2014 `save-temps` option enables users to save all the temporary files in a given directory. Example
`-save-temps=obj -o obj/dir/a.out` will put all temporary files under `obj/dir/`